sdk/storage/src/datalake/file_system_client.cpp

```diff
@@ -0,0 +1,235 @@
+      const std::string& fileSystemUri,
+      const FileSystemClientOptions& options)
+      : m_dfsUri(fileSystemUri)
+  {
+    m_blobUri = Details::GetBlobUriFromDfsUri(m_dfsUri);
+
+    std::vector<std::unique_ptr<Azure::Core::Http::HttpPolicy>> policies;
+    for (const auto& p : options.policies)
+    {
+      policies.emplace_back(std::unique_ptr<Azure::Core::Http::HttpPolicy>(p->Clone()));
+    }
+    policies.emplace_back(std::make_unique<CommonHeadersRequestPolicy>());
+    policies.emplace_back(std::make_unique<Azure::Core::Http::TransportPolicy>(
+        std::make_shared<Azure::Core::Http::CurlTransport>()));
+    m_pipeline = std::make_shared<Azure::Core::Http::HttpPipeline>(policies);
+  }
+
+  PathClient FileSystemClient::GetPathClient(const std::string& path) const
+  {
+    PathClient client = PathClient();
+    auto builder = m_dfsUri;
+    builder.AppendPath(path, true);
+    client.m_dfsUri = std::move(builder);
+    client.m_blobUri = Details::GetBlobUriFromDfsUri(builder);
+    client.m_pipeline = m_pipeline;
+    return client;
+  }
+
+  FileSystemCreateResponse FileSystemClient::Create(const FileSystemCreateOptions& options) const
+  {
+    DataLakeRestClient::FileSystem::CreateOptions protocolLayerOptions;
+    // TODO: Add null check here when Nullable<T> is supported
+    protocolLayerOptions.Properties = SerializeMetadata(options.Metadata);
+    protocolLayerOptions.Timeout = options.Timeout;
+    return DataLakeRestClient::FileSystem::Create(
+        m_dfsUri.to_string(), *m_pipeline, options.Context, protocolLayerOptions);
+  }
+
+  FileSystemDeleteResponse FileSystemClient::Delete(const FileSystemDeleteOptions& options) const
+  {
+    DataLakeRestClient::FileSystem::DeleteOptions protocolLayerOptions;
+    // TODO: Add null check here when Nullable<T> is supported
+    protocolLayerOptions.IfModifiedSince = options.IfModifiedSince;
+    protocolLayerOptions.IfUnmodifiedSince = options.IfUnmodifiedSince;
+    protocolLayerOptions.Timeout = options.Timeout;
+    return DataLakeRestClient::FileSystem::Delete(
+        m_dfsUri.to_string(), *m_pipeline, options.Context, protocolLayerOptions);
+  }
+
+  FileSystemGetMetadataResponse FileSystemClient::GetMetadata(
+      const FileSystemGetMetadataOptions& options) const
+  {
+    DataLakeRestClient::FileSystem::GetPropertiesOptions protocolLayerOptions;
+    // TODO: Add null check here when Nullable<T> is supported
+    protocolLayerOptions.Timeout = options.Timeout;
+    auto result = DataLakeRestClient::FileSystem::GetProperties(
+        m_dfsUri.to_string(), *m_pipeline, options.Context, protocolLayerOptions);
+    return FileSystemGetMetadataResponse{
+        std::move(result.Date),
+        std::move(result.ETag),
+        std::move(result.LastModified),
+        std::move(result.RequestId),
+        std::move(result.Version),
+        DeserializeMetadata(result.Properties),
+        result.NamespaceEnabled == "true" ? true : false};
```

Comment:
why not parse it in protocol layer?

Reply:
See swagger:

>                 "x-ms-namespace-enabled": {
                  "x-ms-client-name": "NamespaceEnabled",
                  "description": "A bool string indicates whether the namespace feature is enabled. If \"true\", the namespace is enabled for the filesystem.",
                  "type": "string"

Reply:
It shows that it is a bool string, so I am currently reluctant to change the type to be bool.

Reply:
It'll be fine if you don't want to change swagger

Reply:
But I kind of feel it's a bug of swagger.